intercept/CMakeLists.txt

```diff
@@ -110,6 +110,9 @@ if(CMAKE_LIBRARY_ARCHITECTURE)
     target_compile_definitions(OpenCL PRIVATE CLINTERCEPT_LIBRARY_ARCHITECTURE="${CMAKE_LIBRARY_ARCHITECTURE}")
 endif()
 
+message(STATUS "CMAKE_INSTALL_LIBDIR is: ${CMAKE_INSTALL_LIBDIR}")
+target_compile_definitions(OpenCL PRIVATE CLINTERCEPT_LIBRARY_DIR="${CMAKE_INSTALL_LIBDIR}")
```

Comment:
It looks like `CMAKE_INSTALL_LIBDIR` is only set after including `GNUInstallDirs`.  Since we haven't included `GNUInstallDirs` yet it won't be set the first time CMake gets run.  You can see this by generating build files into a fresh directory or by looking at the CI logs.  Note, it will be set on subsequent runs regardless.

Couple of options:

1. We can include `GNUInstallDirs` earlier, before these lines.
2. We can set `CLINTERCEPT_LIBRARY_DIR` later in this file, after including `GNUInstallDirs`.

I have a slight preference for (2) as this will be consistent with the CMakeLists.txt for `cliloader.